native-media/app/src/main/java/com/example/nativemedia/NativeMedia.java

```diff
@@ -206,10 +214,19 @@ public void onNothingSelected(AdapterView parent) {
         // initialize button click handlers
 
         // Java MediaPlayer start/pause
-
+        final Activity appActivity = this;
```

Comment:
Rather than defining a final Activity here and potentially holding onto a reference, refer to the Activty as 'NativeMedia.this' in the click listener.